core/history/history.go

```diff
@@ -174,6 +183,252 @@ func (h *history) Versions(id domain.FullID, lastVersionId string, limit int) (r
+			response = append(response, message.Msg)
+		}
+	}
+	return response
+}
+
+func isSuitableChange(message simple.EventMessage) bool {
+	return isDataviewChange(message) ||
+		isDetailsChange(message) ||
+		isRelationsChange(message) ||
+		isBlockPropertiesChange(message) ||
+		isSimpleBlockChange(message) ||
+		isBasicBlockChange(message)
+}
+
+func isDataviewChange(message simple.EventMessage) bool {
+	return message.Msg.GetBlockDataviewRelationDelete() != nil ||
+		message.Msg.GetBlockDataviewSourceSet() != nil ||
+		message.Msg.GetBlockDataviewRelationSet() != nil ||
+		message.Msg.GetBlockDataviewViewSet() != nil ||
+		message.Msg.GetBlockDataviewViewOrder() != nil ||
+		message.Msg.GetBlockDataviewViewDelete() != nil ||
+		message.Msg.GetBlockDataViewObjectOrderUpdate() != nil ||
+		message.Msg.GetBlockDataViewGroupOrderUpdate() != nil ||
+		message.Msg.GetBlockDataviewViewUpdate() != nil ||
+		message.Msg.GetBlockDataviewTargetObjectIdSet() != nil
+}
+
+func isRelationsChange(message simple.EventMessage) bool {
+	return message.Msg.GetObjectRelationsAmend() != nil ||
+		message.Msg.GetObjectRelationsRemove() != nil
+}
+
+func isDetailsChange(message simple.EventMessage) bool {
+	return message.Msg.GetObjectDetailsAmend() != nil ||
+		message.Msg.GetObjectDetailsUnset() != nil
+}
+
+func isBlockPropertiesChange(message simple.EventMessage) bool {
+	return message.Msg.GetBlockSetAlign() != nil ||
+		message.Msg.GetBlockSetChildrenIds() != nil ||
+		message.Msg.GetBlockSetBackgroundColor() != nil ||
+		message.Msg.GetBlockSetFields() != nil ||
+		message.Msg.GetBlockSetVerticalAlign() != nil
+}
+
+func isSimpleBlockChange(message simple.EventMessage) bool {
+	return message.Msg.GetBlockSetTableRow() != nil ||
+		message.Msg.GetBlockSetRelation() != nil ||
+		message.Msg.GetBlockSetText() != nil ||
+		message.Msg.GetBlockSetLink() != nil ||
+		message.Msg.GetBlockSetLatex() != nil ||
+		message.Msg.GetBlockSetFile() != nil ||
+		message.Msg.GetBlockSetDiv() != nil ||
+		message.Msg.GetBlockSetBookmark() != nil
+}
+
+func isBasicBlockChange(message simple.EventMessage) bool {
+	return message.Msg.GetBlockAdd() != nil ||
+		message.Msg.GetBlockDelete() != nil
+}
+
+func (h *history) GetBlocksModifiers(id domain.FullID, versionId string, blocks []*model.Block) ([]*model.ObjectViewBlockModifier, error) {
+	if len(blocks) == 0 {
+		return nil, nil
+	}
+	existingBlocks := lo.SliceToMap(blocks, func(item *model.Block) (string, struct{}) { return item.GetId(), struct{}{} })
+	tree, _, err := h.treeWithId(id, versionId, true)
+	if err != nil {
+		return nil, err
+	}
+
+	blocksModifiersMap := make(map[string]string, 0)
+	err = tree.IterateFrom(tree.Root().Id, source.UnmarshalChange, func(c *objecttree.Change) (isContinue bool) {
+		if !c.IsSnapshot {
+			h.processChange(c, id, blocksModifiersMap, existingBlocks)
+		}
+		return true
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	blocksModifiers := make([]*model.ObjectViewBlockModifier, 0)
```

Comment:
The naming is a bit confusing. We call BlockModifier an identity which modified the block and not some `modifier` of a block like some parameter or something (e.g. layout).

Maybe BlockParticipant?

Reply:
Fixed